pkg/pagerduty/service.go

```diff
@@ -108,22 +108,44 @@ func (data *Data) ParsePDConfig(osc client.Client) error {
 }
 
 // GetService searches the PD API for an already existing service
-func (data *Data) GetService() (string, error) {
+func (data *Data) GetService() (pdApi.Service, error) {
 	client := pdApi.NewClient(data.APIKey)
 	var opts pdApi.ListServiceOptions
 	opts.Query = data.servicePrefix + "-" + data.ClusterID + "." + data.BaseDomain + "-hive-cluster"
 	services, err := client.ListServices(opts)
 	if err != nil {
-		return "", err
+		return pdApi.Service{}, err
 	}
 
 	if len(services.Services) <= 0 {
-		return "", errors.New("No services returned from PagerDuty")
+		return pdApi.Service{}, errors.New("No services returned from PagerDuty")
 	} else if len(services.Services) > 1 {
-		return "", errors.New("Multiple services returned from PagerDuty")
+		return pdApi.Service{}, errors.New("Multiple services returned from PagerDuty")
 	}
 
-	return services.Services[0].ID, nil
+	return services.Services[0], nil
+}
+
+// GetIntegrationKey searches the PD API for an already existing service and returns the first integration key
+func (data *Data) GetIntegrationKey() (string, error) {
```

Comment:
What does accessing the first integration key get us? And if there's more than one, what does that mean; does this method operate on a specific type of integration (eg DeadManSnitch) only?

Reply:
DeadManSnitch will have it's own service. The problem is that integrations can have the same name, so we don't know if it is the proper one, if we throw an error I plan on adding metrics that track the error count so if we detect an issue we can go look. https://i.imgur.com/f0pcfGM.jpg